r-package/grf/tests/testthat/test_lm_forest.R

```diff
@@ -0,0 +1,61 @@
+test_that("lm_forest with single W ~ causal forest", {
+  # Binary W
+  n <- 500
+  p <- 5
+  X <- matrix(rnorm(n * p), n, p)
+  W <- rbinom(n, 1, 0.5)
+  Y <- pmax(X[, 1], 0) * W + X[, 2] + pmin(X[, 3], 0) + rnorm(n)
+  wts <- sample(1:2, n, TRUE)
+
+  Y.hat <- predict(regression_forest(X, Y, num.trees = 500))$predictions
+  cf <- causal_forest(X, Y, W, Y.hat = Y.hat, W.hat = 0.5, sample.weights = wts, num.trees = 250, stabilize.splits = FALSE)
+  lmf <- lm_forest(X, Y, W, Y.hat = Y.hat, W.hat = 0.5, sample.weights = wts, num.trees = 250)
+  expect_lt(mean((predict(cf)$predictions - predict(lmf)$predictions[,,])^2), 0.12)
```

Comment:
This is a pretty loose tolerance for a DGP like this. It'd be reassuring to see something like this with a better tolerance.

Reply:
Oh yes, I kept bumping the tol to make it never fail and accidentally made it not discriminate anything anymore